react/tic-tac-toe/src/app/screens/Game/components/Board/index.js

```diff
@@ -1,19 +1,23 @@
 import React, { Component } from 'react';
+import { func, string, arrayOf } from 'prop-types';
 
 import Square from '../Square';
 
 import styles from './styles.module.scss';
 
 class Board extends Component {
-  renderSquare() {
-    return <Square />;
+  renderSquare(i) {
+    const { squares, onClick } = this.props;
+    return (
+      <Square
+        value={squares[i]}
+        onClick={() => onClick(i)}
```

Comment:
Defini la funcion afuera, sino se crea una nueva funcion cada vez que rendereas un Cuadrado.

Reply:
Pero le tengo q pasar si o si el `i` para saber que cuardrado renderear. Defino una funcion que defina una funcion para evitar que se renderie? algo de este estilo:
```javascript
handleClick = i => this.props.onClick(i);

  renderSquare(i) {
    const { squares } = this.props;
    return (
      <Square
        value={squares[i]}
        onClick={() => this.handleClick(i)}
      />);
  }
```

Reply:
Es el mismo problema porque en cada render se crea la funcion () => this.handleClick(i).
Lo que podes hacer es pasarle al Square que numero de cuadrado es para que el llame al onClick con ese numero. No se que piensa @JoaquinBattilana